### DISCUSSION TITLE
Unable to change -march option when building gcc package for libgcc libraries?

### DISCUSSION THREAD

**Discussion by pl752** [msys2/MINGW-packages#20141] [Q&A] [answered]
Hello, I needed to build a dll file for an old target with Pentium II cpu, it was fairly simple to compile my sources for this target, howewer there is a catch, static libs from libgcc are built by default with -march=pentium4 option, so I wanted to recompile it with -march=pentium2 or --with-cpu=pentium2, but I am copletely stuck, as --with-cpu=pentium2 option in PKGBUILD doen't change the fact of -march=pentium4 option being included for compiling tasks and I haven't succeed in finding. Where should one redefine or add arch selection option in the configs or sources in order to override the default -march=pentium4 (package C:\msys64\home\pl752\mingw-w64-gcc)? Or is there some separate mingw package just for libgcc, where it is easier to achieve? Or is it impossible for some source specific reasons?

**Comment by Biswa96** [comment, marked as answer]
Default compiler flags are set using /etc/makepkg_mingw.conf file. There are some ways you can change those default flags. For example,
1. edit the cflags and cxxflags in /etc/makepkg_mingw.conf file manually in your system.
2. disable default compiler flags with `options=('!buildflags')` in PKGBUILD files individually.
3. change cflags or cxxflags in PKGBUILD file with setting those variables `CFLAGS="something" CXXFLAGS="something"`.

**Reply by pl752** [reply]
Thank you very much for an advice, I will try editing asap

**Reply by pl752** [reply]
Aha, it seems it (temporarily editing /etc/makepkg_mingw.conf) did the trick, now build tasks emit:
`echo GCC_CFLAGS = '-g -march=pentium2 -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -DIN_GCC    -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition  -isystem ./include ' >> tmp-libgcc.mvars`
Thank you!
